hazelcast/src/main/java/com/hazelcast/internal/metrics/ProbeBuilder.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.metrics;
+
+/**
+ * Immutable builder object to register Probes.
+ */
+public interface ProbeBuilder {
+
+    /**
+     * Returns a new ProbeBuilder instance with the given tag added.
+     */
+    ProbeBuilder withTag(String tag, String value);
+
+    /**
+     * Builds the String name for manually registering probes.
+     */
+    String metricName();
```

Comment:
Why is this method needed? Who is going to call this method from the outside?

Reply:
Currently no one. I thought it might be useful to do something special: to use the ProbeBuilder just to build the name and not to register a probe.

Reply:
If it isn't used ---> delete it.